lib/inventory/inventory.go

```diff
@@ -119,6 +124,14 @@ type downstreamHandle struct {
 	senderC      chan DownstreamSender
 	closeContext context.Context
 	cancel       context.CancelFunc
+	// agentMetadataCh is the buffered channel (size 1) where the agent metadata
+	// will be sent to at most once (once it is calculated). For this reason, if
+	// the agent metadata is taken out of the channel and we fail to send it
+	// upstream, then the metadata will never be sent upstream as there is no
+	// retry mechanism. However, since the agent metadata is sent after the
+	// hello exchange succeeds, it is unlikely that the sending of the metadata
+	// will fail.
+	agentMetadataCh chan proto.UpstreamInventoryAgentMetadata
```

Comment:
The way this seems to be coded is that we do one attempt to send the metadata per lifetime of the `TeleportProcess`, we should be sending the metadata once per connection to an auth server instead.

Reply:
Yes, this is intentional.

The initial proposal in the RFD would send 1 `UpstreamInventoryAgentMetadata` per `UpstreamInventoryHello`.
However, this complicates the implementation a bit without any clear benefit (as most agent metadata won't change throughout the lifetime of the agent).

Reply:
I'm not sure I like that, it's gonna be far too easy to lose track of the data like that, and in the current model the auth server would be forced to persist the data somewhere if we wanted to keep it available for the user.

Reply:
The easiest way to send 1 `UpstreamInventoryAgentMetadata` per `UpstreamInventoryHello` would be to spawn the goroutine for each new stream: fe4f2790be2db87c1aea1271a037ef72072f91bf

WDYT? Could this spawn many unnecessary goroutines?

---

Alternatively, (as proposed initially in the RFD: https://github.com/gravitational/teleport/commit/7737d46912a990f37c10e2778c21a3dc1a52af02#diff-dbf83bde2c5bcf10c7f0b59b4ed0f2720ff90e8d0ca33f1f4ad2702344299f01) we could spawn a single that:
- computes the metadata just once, and
- sends the metadata through a channel when it is requested